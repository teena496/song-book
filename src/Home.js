import {
  Container,
  TableContainer,
  TableCell,
  TableRow,
  TableBody,
  TableHead,
  Table,
  Paper,
  Button,
} from "@mui/material";

import React, { useEffect, useState } from "react";
import { db } from "./firebase-config";
import {
  collection,
  getDocs,
  doc,
  deleteDoc,
  updateDoc,
} from "firebase/firestore";

export default function Home() {
  const [songs, setSongs] = useState([]);
  const columnNames = [
    { id: 1, name: "Song Name" },
    { id: 2, name: "Artist" },
    { id: 3, name: "Genre" },
    { id: 4, name: "Released" },
    { id: 5, name: "" },
    { id: 6, name: "" },
  ];
  const songsCollection = collection(db, "songs");

  useEffect(() => {
    const getSongs = async () => {
      const data = await getDocs(songsCollection);
      if (data) {
        setSongs(
          data.docs.map((doc) => ({
            ...doc.data(),
            id: doc.id,
          }))
        );
      } else return;
    };
    getSongs();
  }, [songs, songsCollection]);

  const onDeleteSong = async (id) => {
    const songDoc = doc(db, "songs", id);
    deleteDoc(songDoc);
  };

  const onUpdateSong = async (id) => {
    // const songDoc = doc(db, "songs", id);
    // await updateDoc(songDoc, {
    //   name: "Test by Teena",
    // });
  };

  return (
    <div>
      <Container sx={{ padding: "30px" }}>
        <TableContainer component={Paper}>
          <Table sx={{ minWidth: 650 }} aria-label="simple table">
            <TableHead>
              <TableRow>
                {columnNames.map((column) => (
                  <TableCell
                    key={`columnname-${column.id}`}
                    data-testid={`columnname-${column.id}`}
                    sx={{ fontSize: "1rem" }}
                  >
                    {column.name}
                  </TableCell>
                ))}
              </TableRow>
            </TableHead>
            <TableBody>
              {songs.map((row) => (
                <TableRow>
                  <TableCell>
                    <a href={row.link}>{row.name}</a>
                  </TableCell>
                  <TableCell>{row.artist}</TableCell>
                  <TableCell>{row.genre}</TableCell>
                  <TableCell>{row.released}</TableCell>
                  <TableCell>
                    <Button onClick={() => onDeleteSong(row.id)}>Delete</Button>
                  </TableCell>
                  <TableCell>
                    <Button onClick={() => onUpdateSong(row.id)}>Update</Button>
                  </TableCell>
                </TableRow>
              ))}
            </TableBody>
          </Table>
        </TableContainer>
      </Container>
    </div>
  );
}
