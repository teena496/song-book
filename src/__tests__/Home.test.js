import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import Home from "../Home";

test("should render Home component", () => {
  render(<Home />);
  const columnSongName = screen.getByTestId("columnname-1");
  console.log(columnSongName);
  expect(columnSongName.textContent).toBe("Song Name");
});

it("matches snapshot", () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});
