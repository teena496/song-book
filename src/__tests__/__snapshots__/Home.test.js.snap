// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-qfxye4-MuiContainer-root"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
    >
      <table
        aria-label="simple table"
        className="MuiTable-root css-quj9j6-MuiTable-root"
        role={null}
      >
        <thead
          className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1badeqd-MuiTableCell-root"
              data-testid="columnname-1"
              scope="col"
            >
              Song Name
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1badeqd-MuiTableCell-root"
              data-testid="columnname-2"
              scope="col"
            >
              Artist
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1badeqd-MuiTableCell-root"
              data-testid="columnname-3"
              scope="col"
            >
              Genre
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1badeqd-MuiTableCell-root"
              data-testid="columnname-4"
              scope="col"
            >
              Released
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1badeqd-MuiTableCell-root"
              data-testid="columnname-5"
              scope="col"
            />
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1badeqd-MuiTableCell-root"
              data-testid="columnname-6"
              scope="col"
            />
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          role={null}
        />
      </table>
    </div>
  </div>
</div>
`;
